packages/dbos-cloud/users/login.ts

```diff
@@ -45,3 +61,202 @@ export async function login(host: string, getRefreshToken: boolean, useRefreshTo
+    return emptyCredentials;
+  }
+  const credentials = JSON.parse(fs.readFileSync(`./${dbosEnvPath}/credentials`).toString("utf-8")) as DBOSCloudCredentials;
+  credentials.token = credentials.token.replace(/\r|\n/g, ""); // Trim the trailing /r /n.
+  logger.debug(`Loaded credentials from ${dbosEnvPath}/credentials`);
+  if (isTokenExpired(credentials.token)) {
+    if (credentials.refreshToken) {
+      logger.debug("Refreshing access token with refresh token");
+      const authResponse = await authenticateWithRefreshToken(logger, credentials.refreshToken);
+      if (authResponse === null) {
+        logger.warn("Refreshing access token with refresh token failed. Logging in again...");
+        deleteCredentials();
+        return emptyCredentials;
+      } else {
+        // Update the token and save the credentials
+        credentials.token = authResponse.token.replace(/\r|\n/g, "");
+      }
+    } else {
+      logger.warn("Credentials expired. Logging in again...");
+      deleteCredentials();
+      return emptyCredentials;
+    }
+  }
+  return credentials;
+}
+
+/**
+ * Check user profile in DBOS Cloud.
+ * @param host - The DBOS Cloud host to authenticate against.
+ * @param credentials - The user's DBOS Cloud credentials (to be updated with userName and organization).
+ * @param logger  - The logger instance.
+ * @returns {boolean} - True if the user profile exists, false otherwise.
+ */
+async function checkUserProfile(host: string, credentials: DBOSCloudCredentials, logger: Logger): Promise<boolean> {
+  const bearerToken = "Bearer " + credentials.token;
+  try {
+    const response = await axios.get(`https://${host}/v1alpha1/user/profile`, {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: bearerToken,
+      },
+    });
+    const profile = response.data as UserProfile;
+    credentials.userName = profile.Name;
+    credentials.organization = profile.Organization;
+    return true;
+  } catch (e) {
+    const axiosError = e as AxiosError;
+    const errorLabel = `Failed to login`;
+    if (isCloudAPIErrorResponse(axiosError.response?.data)) {
+      const resp: CloudAPIErrorResponse = axiosError.response?.data;
+      if (!resp.message.includes("user not found in DBOS Cloud")) {
+        handleAPIErrors(errorLabel, axiosError);
+        process.exit(1);
+      }
+    } else {
+      logger.error(`${errorLabel}: ${(e as Error).message}`);
+      process.exit(1);
+    }
+  }
+  return false;
+}
+
+/**
+ * Register the user in DBOS Cloud and modify the credentials.
+ * Exit the process on errors.
+ * @param {string} host - The DBOS Cloud host to authenticate against.
+ * @param {DBOSCloudCredentials} credentials - The user's DBOS Cloud credentials (to be updated with userName and organization).
+ * @param {Logger} logger - The logger instance.
+ * @returns
+ */
+async function registerUser(host: string, credentials: DBOSCloudCredentials, logger: Logger): Promise<void> {
```

Comment:
Why do we now have two implementations of register, one of which is in `login.ts`?

Reply:
Updated register.ts